src/App.js

```diff
@@ -1,27 +1,76 @@
-import React from 'react'
+import React, { useState } from 'react'
 import Nav from './Nav'
 import Homepage from './Homepage'
 import Login from './Login'
 import About from './About'
 
 import { Route, Switch } from 'react-router-dom';
+import { gql, useQuery, useMutation } from '@apollo/client';
+
 const App = () => {
 
+const [user, setUser] = useState({})
+
+const [username, setUsername] = useState('')
+const [email, setEmail] = useState('')
+
+
+const CREATE_USER = gql`
+
+mutation createUser(
+  $username: String!
+  $email: String!
+  ) {
+  createUser( input: {
+    username: $username
+    email: $email
+  } ) {
+    user {
+    id
+    username
+    email
+    }
+  }
+}
+
+`
+
+const [createUser, { data, error, loading }] = useMutation(CREATE_USER)
+
+const loginUser = (username, email) => {
+  setUsername(username)
+  setEmail(email)
+
+  createUser({
+    variables: {
+      username: username,
+      email: email
+    }
+  })
+  .then((data) => setUser(data.data.createUser.user))
+
+
+  if (error) {
+    console.log(error)
+  }
+}
+
+
 return (
   <div>
     <Switch>
       <Route exact path='/'>
-        <Login />
+        <Login loginUser={loginUser}/>
```

Comment:
you might consider just calling this `user` by definition of the component the user is already logged in.

user is less to type and when you start tot use that Var in the app it's always easier to type user vs loginUser